### ISSUE TITLE
RegExp Replace stopped if 998+ matches found

### ISSUE THREAD

**Issue by SilverBut** [Microsoft/vscode#34517]
- Version 1.16.1
- OS Version: Linux  4.12.13-1-ARCH #1 SMP PREEMPT Fri Sep 15 06:36:43 UTC 2017 x86_64 GNU/Linux
- Commit 27492b6bf3acb0775d82d2f87b25a93490673c6d
- Date 2017-09-14T16:26:20.018Z
- Shell 1.7.3
- Renderer 58.0.3029.110
- Node 7.9.0
- Architecture x64

Steps to Reproduce:

1. Build this test document in a new window(just kept ctrl+c and ctrl+v)
```
a b c 10
<Repeat last line for 999 times>
```
2. Ctrl+H to call Replacement, and input this as your regexp to replace from, and replace it to 123123:
```
^(\S+)\s+?(.+)\s+?(\d+)\d*$
```

3. Replace for all

Excepted Result:

All lines should be replaced to 123123

Produced Result:

Nothing happened.

Futher Steps:

4. Change the first line to `dddddd`

5. Replace use those arguments again

Produced Result:

All lines except the first line is replace to `123123`

Reproduces without extensions: Yes

**Comment by SilverBut** [comment]
Seems related with #32522, #33573, #33989 and #11973.

One of them is submitted in 29 days ago? Really? For a bug in a common function?

**Comment by alexandrudima** [comment]
Duplicate of #32522 i.e. 54ca5101065a7cdb73049755e0393e7b2b31477f also fixes this.

**Comment by SilverBut** [comment]
@alexandrudima Seems not fixed, at least at this version. Have not released yet?

Version 1.16.1
Commit 27492b6bf3acb0775d82d2f87b25a93490673c6d
Date 2017-09-14T16:26:20.018Z
Shell 1.7.3
Renderer 58.0.3029.110
Node 7.9.0
Architecture x64

**Comment by alexandrudima** [comment]
@SilverBut  The fix should come with 1.17 .
